### PR TITLE
gh-103650: Fix perf maps address format

### LINKED ISSUE #103650: perf maps address format
# Bug report

As implemented by @pablogsal in #96123 we can now generate perf maps(which is awesome BTW).
Currently, perf maps are formatted this way:
```
0x58c5882 b py::<module>:/home/arthur/Projects/python-playground/dummy_script.py
0x58c588d b py::baz:/home/arthur/Projects/python-playground/dummy_script.py
0x58c5898 b py::bar:/home/arthur/Projects/python-playground/dummy_script.py
0x58c58a3 b py::foo:/home/arthur/Projects/python-playground/dummy_script.py
```

However, as per the [perf maps interface specification](https://github.com/torvalds/linux/blob/master/tools/perf/Documentation/jit-interface.txt) it seems the address should not include the leading `0x`. 

Thus the perf map should look like this:
```
58c5882 b py::<module>:/home/arthur/Projects/python-playground/dummy_script.py
58c588d b py::baz:/home/arthur/Projects/python-playground/dummy_script.py
58c5898 b py::bar:/home/arthur/Projects/python-playground/dummy_script.py
58c58a3 b py::foo:/home/arthur/Projects/python-playground/dummy_script.py
```
 
Still, it works fine with `perf` but might not with other tools not handling this leading `0x`.

# Your environment

- CPython versions tested on: 3.12.0a7
- Operating system and architecture: Manjaro 22.1.0 x86_64

### Linked PRs
* gh-103651

### PR DESCRIPTION
**Pull request by art049**
* Issue: gh-103650

### CODE REVIEW

File: Python/perf_trampoline.c [hunk 1]
```diff
@@ -253,7 +253,7 @@ perf_map_write_entry(void *state, const void *code_addr,
                                   NULL);
         return;
     }
-    fprintf(method_file, "%p %x py::%s:%s\n", code_addr, code_size, entry,
+    fprintf(method_file, "%llx %x py::%s:%s\n", code_addr, code_size, entry,
```
Comment (sunmy2019) [inline Python/perf_trampoline.c]: What if it's a 32-bit machine? Python still supports that https://www.python.org/downloads/release/python-3120a7/

Also, the compiler reports a lot of warnings. This should be fixed.

**Review by sunmy2019 — CHANGES_REQUESTED** [review]

File: Python/perf_trampoline.c (hunk 1 above)
Comment (sunmy2019) [reply Python/perf_trampoline.c]: %p is an implementation-defined format specifier; the c11 specs don't clarify if it contains "0x"

Comment (art049) [reply Python/perf_trampoline.c]: I casted to `uintptr_t` and used `PRIxPTR` to handle the format. This fixed compiler warnings and should work both on 32 and 64-bit machines.

**Comment by terryjreedy** [comment]
I think that this should wait until Pablo verifies that it is indeed the implementation that should be changed.

**Comment by carljm** [comment]
@art049 did you verify that the new output after this PR works the same with `perf`?

@terryjreedy since the relevant docs here are the external perf map specification, not the CPython docs, it seems clear that the implementation in CPython should match the specification. There is no option to change the specification, we don't own it.

**Comment by pablogsal** [comment]
Running this PR with the buildbots as the perf stuff is not checked on regular CI.

**Review by pablogsal — APPROVED** [review]

**Pull request merged**